manifests/init.pp

```diff
@@ -56,21 +58,25 @@
     }
   }
 
-  package { 'ipmitool':
-    ensure => $package_ensure,
-    name   => 'ipmitool',
-  }
+  if $manage_package {
+    package { 'ipmitool':
+      ensure => $package_ensure,
+      name   => 'ipmitool',
+    }
 
-  package { 'ipmidriver':
-    ensure => $package_ensure,
-    name   => $openipmi,
+    package { 'ipmidriver':
+      ensure => $package_ensure,
+      name   => $openipmi,
+    }
   }
 
-  service { $service_name:
-    ensure     => $service_ensure,
-    enable     => true,
-    hasrestart => true,
-    hasstatus  => true,
-    require    => [ Package['ipmitool'], Package['ipmidriver']],
+if $manage_service {
+    service { $service_name:
+      ensure     => $service_ensure,
+      enable     => true,
+      hasrestart => true,
+      hasstatus  => true,
+      subscribe  => [ Package['ipmitool'], Package['ipmidriver'], File['ipmiconfig'] ],
```

Comment:
Actually this is a problem here too.  Because we can selectively manage the package, the service resource would fail if we decided to manage the service but not the package and vice versa even if we used notify statements.  We can't notify or subscribe to a resource that does not exist in the catalog.